up42/auth.py

```diff
@@ -144,10 +171,39 @@ def _endpoint(self) -> str:
         return f"https://api.up42.{self.env}"
 
     def _get_token(self):
+        if self.auth_type == AuthType.PROJECT.value:
+            self._get_token_project_based()
+        elif self.auth_type == AuthType.ACCOUNT.value:
+            self._get_token_account_based()
+
+    def _get_token_account_based(self):
+        """Account based authentication via username and password."""
+        try:
+            req_headers = {
+                "Content-Type": "application/x-www-form-urlencoded",
+            }
+            req_body = {
+                "grant_type": "password",
+                "username": self.credentials_id,
+                "password": self.credentials_key,
+            }
+            token_response = requests.post(
+                url=self._endpoint() + "/oauth/token",
+                data=req_body,
+                headers=req_headers,
+                timeout=1200,
+            )
+            if token_response.status_code != 200:
+                raise ValueError("Authentication was not successful, check the provided credentials.")
+        except requests.exceptions.RequestException as err:
```

Comment:
we are raising the same error message twice.
Can we combine the two?

Reply:
Not combine but raise more appropriate errors. First was for a not valid credentials the later is for network errors.

Reply:
of-course! 🤦🏻